src/WebAsmReader.h

```diff
@@ -0,0 +1,124 @@
+#ifndef WEBASMREADER_H
+#define WEBASMREADER_H
+
+
+#include <sys/stat.h>
+#include "support/file.h"
+#include "wasm-binary.h"
+#include "wasm-s-parser.h"
+#include "wasm-printing.h"
+
+using namespace cashew;
+using namespace wasm;
+
+struct WebAsmReader
+{
+  
+  bool m_DebugInfo;
+  bool m_validateWeb;
+  WebAsmReader(bool debugInfo=false,bool valWeb=false):
+             m_DebugInfo(debugInfo),m_validateWeb(valWeb){}
+                                      
+  void write(std::string &inputFileName, 
+             std::string &outputFileName,
+             bool debug=false)
+  {
+    if (inputFileName.length()>5)
+    {
+      auto ext = inputFileName.substr(inputFileName.length()-5);
+      if (ext.compare(".wasm")==0)
+      {
+        writeWast(inputFileName,outputFileName,debug);
+        return;
+      }
+      if (ext.compare(".wast")==0)
+      {
+        writeWasm(inputFileName,outputFileName);
+       
+        //m_textInput=read_file<std::string>(fileName, Flags::Text, debug ? Flags::Debug : Flags::Release);
+        
+        return;
+      }
+    }
+    
+    throw ParseException("filename must end in .wast or .wasm");
```

Comment:
not sure about this, but perhaps we should tolerate other suffixes, and default them to either text or binary. currently `check.py` will emit some other suffixes, which I guess we could change, but it might be simpler not to. Defaulting to text would be the simplest, as it's what `check.py` does.